pulsar/internal/compression/zstd_cgo.go

```diff
@@ -61,6 +64,8 @@ func (z *zstdCGoProvider) CompressMaxSize(originalSize int) int {
 }
 
 func (z *zstdCGoProvider) Compress(dst, src []byte) []byte {
+	z.mu.Lock()
+	defer z.mu.Unlock()
```

Comment:
If there are many routine publishing, we could get into a contention situation. Maybe we could try some gorouting-local access? eg: https://github.com/timandy/routine?tab=readme-ov-file#use-threadlocal

Reply:
Alternatively, golang provides [sync.Pool](https://pkg.go.dev/sync#Pool) which could potentially be used to create and cache zstd contexts as required.

Reply:
Good point, the `sync.Pool` is already doing the thread-local caching internally.

Reply:
Good idea! Thanks.